### PR TITLE
Improved parsing of class II HLA-DRB alleles

### PR DESCRIPTION
**Pull request by iskandr**
When an allele is specified as "HLA-DRB1", it's implicitly assuming the alpha chain is "DRA1\*01:01". Previously "HLA-DRA1\*01:01-DRB1*01:01" would not have normalized to the same string as "HLA-DRB1\*01:01". 

Also, made a `normalization` module which contains `normalize_allele_name` and `compact_allele_name`. Looked in mhctools and a few other downstream tools and it does not seem to break their usage pattern to move these functions.

### CODE REVIEW

**Comment by tavinathanson** [comment]
@iskandr re "Previously "HLA-DRA1*01:01-DRB1*01:01" would not have normalized to the same string as "HLA-DRB1*01:01"": what did it normalize to before this?

**Comment by tavinathanson** [comment]
@iskandr oh, I see, you're saying it just normalized to "HLA-DRB1*01:01"?

**Comment by iskandr** [comment]
@tavinathanson Yeah, the parsing code was very naive in that if the original string had a DRA gene that would end up in the normalized form and otherwise it would just have the DRB component. 

Eventually we need to significantly improve the parsing by explicitly enumerating all the species/gene pairs, along with their synonyms and when alpha or beta chains of Class II names can be omitted by convention. Trying to pull a normalized form out of just the input string without domain knowledge is never going to be a 100% solution.

File: mhcnames/normalization.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    Turn HLA-A*02:01 into A0201 or H-2-D-b into H-2Db or
+    HLA-DPA1*01:05-DPB1*100:01 into DPA10105-DPB110001
+    """
+    parsed_alleles = parse_classi_or_classii_allele_name(raw_allele)
+    normalized_list = []
+    if len(parsed_alleles) == 2:
+        alpha, beta = parsed_alleles
+        # by convention the alpha allelle is omitted since it's assumed
+        # to be DRA1*01:01
+        if alpha == _DRA1_0101:
```
Comment (tavinathanson) [inline mhcnames/normalization.py:103]: Does anything test this?

File: test/test_human_class2.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 )
 
 def test_human_class2():
-    expected = "HLA-DRB1*01:02"
+    expected = "HLA-DRA1*01:01-DRB1*01:02"
     expected_compact = "DRB10102"
     for name in ["DRB1_0102",
```
Comment (tavinathanson) [inline test/test_human_class2.py:11]: Should this test also include `HLA-DRA1*01:01-DRB1*01:02` in the names list?

File: mhcnames/normalization.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+_DRA1_0101 = AlleleName(
+    species="HLA",
+    gene="DRA1",
+    allele_family="01",
+    allele_code="01")
+
+def compact_allele_name(raw_allele):
+    """
+    Turn HLA-A*02:01 into A0201 or H-2-D-b into H-2Db or
+    HLA-DPA1*01:05-DPB1*100:01 into DPA10105-DPB110001
```
Comment (tavinathanson) [inline mhcnames/normalization.py:95]: Mention here that compact is "the smallest repr that still uniquely identifies the allele given some context"?

File: test/test_human_class2.py (hunk 2 above)
Comment (tavinathanson) [reply test/test_human_class2.py:11]: I see that format is tested below, but not with 0101 alpha?

File: mhcnames/allele_name.py [hunk 4]
```diff
@@ -128,6 +92,13 @@ def parse_allele_name(name, species_prefix=None):
         # MHC class II genes like "DQA1" need to be parsed with both
         # letters and numbers
         gene, name = parse_alphanum(name, 4)
+        # TODO: make a list of known species/gene pairs, along with
+        # gene synonyms. That should significantly imporve on this kind of
+        # ad-hoc synonym handling.
+        if gene == "DRA":
```
Comment (tavinathanson) [inline mhcnames/allele_name.py:98]: Is this tested?

File: .travis.yml [hunk 5]
```diff
@@ -2,7 +2,7 @@ sudo: false  # Use container-based infrastructure
 language: python
 python:
   - "2.7"
-  - "3.4"
+  - "3.5"
```
Comment (tavinathanson) [inline .travis.yml:5]: Curious why? Should we do this across repos?

**Review by tavinathanson — COMMENTED** [review]
LGTM % some comments/questions

File: .travis.yml (hunk 5 above)
Comment (iskandr) [reply .travis.yml:5]: I don't know if it matters, I just did it to see if it changes the test results on Travis (which it doesn't). 

Maybe worth standardizing on a newer release, 3.5 seems to have pretty broad adoption.

**Pull request merged**